api/git/types/types.go

```diff
@@ -1,10 +1,19 @@
 package gittypes
 
+// RepoConfig represents a configuration for a repo
 type RepoConfig struct {
 	// The repo url
-	URL string `example:"https://github.com/portainer/portainer-ee.git"`
+	URL string `example:"https://github.com/portainer/portainer.git"`
 	// The reference name
 	ReferenceName string `example:"refs/heads/branch_name"`
 	// Path to where the config file is in this url/refName
 	ConfigFilePath string `example:"docker-compose.yml"`
+	// Git credentials
+	Authentication *GitAuthentication
+	ConfigHash     string
```

Comment:
Didn't know all these comments are used for swagger.
Btw, checking published schema I don't think they actually propagate: https://documentation.portainer.io/api/api-schema/
either because the gitconfig field type is a reference or maybe it doesn't build the whole type tree and stops on the top type.